centralized-sampling-tests/integration-tests/src/main/java/com.amazon.tests/CentralizedSamplingIntegrationTests.java

```diff
@@ -0,0 +1,313 @@
+      return false;
+    }
+    return true;
+  }
+
+  /**
+   * Function to call xray backend to create a sampleRule
+   *
+   * @param jsonBody - JSONBody of the sampleRule
+   * @param ruleName - name of the Rule
+   * @throws IOException - throws if unable to connect to xray backend
+   */
+  public static void makeRule(String jsonBody, String ruleName) throws IOException {
+    // Default rule exists in x-ray by default hence the name
+    if (ruleName.equals("Default")) {
+      return;
+    }
+    System.out.println("Creating " + ruleName + " sample rule");
+    MediaType json = MediaType.get("application/json; charset=utf-8");
+
+    RequestBody reqbody = RequestBody.create(json, jsonBody);
+    try (Response response =
+        httpClient()
+            .newCall(
+                new Request.Builder()
+                    .url("http://localhost:2000/CreateSamplingRule")
+                    .method("POST", reqbody)
+                    .build())
+            .execute()) {
+    } catch (IOException e) {
+      throw new IOException("Could not fetch endpoint", e);
+    }
+  }
+
+  /**
+   * Function to call xray backend to delete a sampleRule
+   *
+   * @param ruleName - name of the Rule
+   * @throws IOException - throws if unable to connect to xray backend
+   */
+  public static void deleteRule(String ruleName) throws IOException {
+    System.out.println("Deleting " + ruleName + " sample rule");
+
+    MediaType json = MediaType.get("application/json; charset=utf-8");
+    String jsonBody = "{\n" + "   \"RuleName\": \"" + ruleName + "\"\n" + "}\n";
+    RequestBody reqbody = RequestBody.create(json, jsonBody);
+    try (Response response =
+        httpClient()
+            .newCall(
+                new Request.Builder()
+                    .url("http://localhost:2000/DeleteSamplingRule")
+                    .method("POST", reqbody)
+                    .build())
+            .execute()) {
+    } catch (IOException e) {
+      throw new IOException("Could not fetch endpoint", e);
+    }
+  }
+
+  /**
+   * Function to run tests on Reservoir SampleRules. Creates one rule at a time, waits 20s for
+   * reservoir to be set runs all test cases, then deletes the rule. Does this for all reservoir
+   * sample rules
+   *
+   * @throws IOException throws if unable to connect to xray backend
+   * @throws InterruptedException if a test fails after retries
+   */
+  public static void reservoirTests() throws IOException, InterruptedException {
+    SampleRule[] sampleRules = SampleRules.getReservoirRules();
+    for (SampleRule sampleRule : sampleRules) {
+      try {
+        makeRule(sampleRule.getJson(), sampleRule.getName());
+      } catch (IOException exception) {
+        System.out.println("Could not fetch endpoint, XRay backend might not be running");
+        throw new IOException();
+      }
+      boolean passed = false;
+      for (int j = 0; j < GenericConstants.MAX_RETRIES; j++) {
```

Comment:
There might be some elegant retry framework or code style we can refer to https://stackoverflow.com/questions/4738510/retry-task-framework